### PR TITLE
Feature#7 admin design reports

### PR DESCRIPTION
**Pull request by xManuel007**
# Description

In this issue we created the view of Reports and add a gem to generate fake data just for have something to analyze
we added, You can see the stats by Company
- Attendances chart
- Average attendances chart
- Abscenses chart
The Gem Faker can create fake information so we create method in the seed file to pregenerate data.

Examples:
To generate a random full name of a company you just have to use the gem like this, also with address, names and a lot of
documentation, its very useful 

image

https://github.com/faker-ruby/faker

## Screenshots

Results of a company
image

image

image

### CODE REVIEW

**Review by MaxRetana — APPROVED** [review]
LGTM

**Review by AangelF — APPROVED** [review]

**Review by Reiyirak — APPROVED** [review]
LGTM

File: app/controllers/admin_reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class AdminReportsController < ApplicationController
+  helper_method :attendance
+  helper_method :average
+  helper_method :absence
+  # helper_method :options
```
Comment (JuanequeX) [inline app/controllers/admin_reports_controller.rb:5]: Delete this comment

File: app/controllers/admin_reports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class AdminReportsController < ApplicationController
+  helper_method :attendance
+  helper_method :average
+  helper_method :absence
```
Comment (JuanequeX) [inline app/controllers/admin_reports_controller.rb:4]: change for this `helper_method :attendance, :average, :absence`

File: app/controllers/admin_reports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def attendance
+    checks = Check.where(employee_id: company.employees.ids)
+    checks.group_by_day(:datetime).where(check: 'in').count
+  end
+
+  def average
+      checks = Check.where(employee_id: company.employees.ids)
+      checks.group_by_month(:datetime).average(:employee_id)
```
Comment (JuanequeX) [inline app/controllers/admin_reports_controller.rb:23]: indented correctly this block

File: app/helpers/admin_reports_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AdminReportsHelper
+end
```
Comment (JuanequeX) [inline app/helpers/admin_reports_helper.rb:2]: if this helper are unnecessary please delete it

File: app/javascript/application.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
 // Configure your import map in config/importmap.rb. Read more: https://github.com/rails/importmap-rails
 import "@hotwired/turbo-rails"
 import "controllers"
+import "chartkick"
+import "Chart.bundle"
```
Comment (JuanequeX) [inline app/javascript/application.js:5]: Add a blank line here

File: app/javascript/clock.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+setInterval(time,1000);
+
+document.addEventListener("DOMContentLoaded", function(){
+	var alert = document.querySelector('.alert');
+	if (alert){
+		setTimeout(function(){
+			alert.style.display = "none";
+		}, 5000);
+	}
+});
```
Comment (JuanequeX) [inline app/javascript/clock.js:18]: Add a blank line here

File: app/views/admin_reports/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<a href="<%= admin_index_path %>" class="a">
```
Comment (JuanequeX) [inline app/views/admin_reports/index.html.erb:1]: class `a`?, these class name not have an descriptive approach

File: config/importmap.rb [hunk 8]
```diff
@@ -4,4 +4,6 @@
 pin '@hotwired/turbo-rails', to: 'turbo.min.js', preload: true
 pin '@hotwired/stimulus', to: 'stimulus.min.js', preload: true
 pin '@hotwired/stimulus-loading', to: 'stimulus-loading.js', preload: true
-pin_all_from 'app/javascript/controllers', under: 'controllers'
\ No newline at end of file
+pin_all_from 'app/javascript/controllers', under: 'controllers'
+pin "chartkick", to: "chartkick.js"
+pin "Chart.bundle", to: "Chart.bundle.js"
```
Comment (JuanequeX) [inline config/importmap.rb:9]: add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/base/_resets.scss [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+asi cuando ponga un <button> directamente ya tendra un estilo base y ahorraremos tiempo
+poniendo estilo y clases :) */
+@import '../utilities/variables';
+
+button {
+  border: none;
+  text-shadow: 0.5px 1px 1px $--shadow-text-color;
+
+  &:hover {
+    background-color: lightgrey;
```
[the remark this answers is not in the document]
Comment (xManuel007) [reply app/assets/stylesheets/base/_resets.scss:12]: done

File: app/assets/stylesheets/pages/_adminReports.scss [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    padding: 0.5rem;
+  }
+
+  &__content {
+    align-items: center;
+    background-color: $background-color;
+    display: flex;
+    margin: 2rem;
+    // width: 90rem;
+    // height: 30rem;
```
[the remark this answers is not in the document]
Comment (xManuel007) [reply app/assets/stylesheets/pages/_adminReports.scss:31]: done

File: app/controllers/admin_reports_controller.rb (hunk 1 above)
Comment (xManuel007) [reply app/controllers/admin_reports_controller.rb:5]: done

File: app/controllers/admin_reports_controller.rb (hunk 2 above)
Comment (xManuel007) [reply app/controllers/admin_reports_controller.rb:4]: ty, done

**Review by JuanequeX — APPROVED** [review]
NIce bandamax

**Pull request merged**